src/journal/journald-server.c

```diff
@@ -846,7 +847,10 @@ static void dispatch_message_real(
                 }
 #endif
 
-                r = cg_pid_get_path_shifted(ucred->pid, s->cgroup_root, &c);
+                if (cgroup)
+                        c = cgroup, r = 0;
```

Comment:
I'm not sure how I can move it into the else branch.  I just wanted a minimal change - what about making the comma expression separate lines?